### PR TITLE
GAM-8 Added Modals and improved form and controller code

### PR DESCRIPTION
**Pull request by Christian-Rades**

### CODE REVIEW

File: app/Resources/views/modals/modal.html.twig [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<div class="modal-header">
+    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+        <span aria-hidden="true">&times;</span>
+    </button>
+    <h4 class="modal-title">{% block modalTitle %}{% endblock %}</h4>
+</div>
+
+    {% block modalbegin %}
```
Comment (tobiasberge) [inline app/Resources/views/modals/modal.html.twig]: Please fix indentation

File: app/Resources/views/modals/editPlayer.html.twig [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <button class="btn btn-default" data-dismiss="modal">Abort</button>
+    {{ form_widget(form.delete,{'attr': {'class': 'btn-danger'}}) }}
+    {{ form_widget(form.save, {'attr': {'class': 'btn-primary'}}) }}
+{% endblock %}
+
+{% block modalend %}
+    {{ form_end(form) }}
+{% endblock %}
+
+
```
Comment (tobiasberge) [inline app/Resources/views/modals/editPlayer.html.twig]: Please remove line breaks

File: src/AppBundle/Controller/GameController.php [hunk 3]
```diff
@@ -41,7 +41,9 @@ public function scoresAction($game_id)
     public function createAction(Request $request)
     {
         $game = new Game();
-        $form = $this->createForm(new GameForm(), $game);
+        $form = $this->createForm(new GameForm(), $game, array(
```
Comment (dbeier) [inline src/AppBundle/Controller/GameController.php]: use [ and not array

array ('bla' => 'blub') --> ['bla' => 'blub']

File: src/AppBundle/Controller/GameController.php [hunk 4]
```diff
@@ -52,23 +54,27 @@ public function createAction(Request $request)
...
     /**
-     * @Route("/edit/{name}")
+     * @Route("/edit/{id}", name="editGame")
      * @return \Symfony\Component\HttpFoundation\Response
      **/
-    public function editAction(Request $request, $name)
+    public function editAction(Request $request, $id)
     {
         $em = $this->getDoctrine()->getManager();
-        $game = $em->getRepository('AppBundle:Game')->findOneByname($name);
```
Comment (dbeier) [inline src/AppBundle/Controller/GameController.php]: inject repository

File: src/AppBundle/Controller/GameController.php [hunk 5]
```diff
@@ -41,7 +41,9 @@ public function scoresAction($game_id)
     public function createAction(Request $request)
     {
         $game = new Game();
-        $form = $this->createForm(new GameForm(), $game);
+        $form = $this->createForm(new GameForm(), $game, array(
+            'action' => $this->get('router')->generate('createGame')
```
Comment (dbeier) [inline src/AppBundle/Controller/GameController.php]: inject router

File: src/AppBundle/Controller/LeagueController.php [hunk 6]
```diff
@@ -23,12 +23,14 @@ public function overviewAction()
...
     /**
-     * @Route("/")
+     * @Route("/create", name="createLeague")
      */
     public function createAction(Request $request)
     {
         $league = new League();
-        $form = $this->createForm(new LeagueForm(), $league);
+        $form = $this->createForm(new LeagueForm(), $league,array(
+            'action' => $this->get('router')->generate('createLeague')
```
Comment (dbeier) [inline src/AppBundle/Controller/LeagueController.php]: inject router

File: src/AppBundle/Controller/LeagueController.php [hunk 7]
```diff
@@ -54,7 +56,11 @@ public function editAction(Request $request, $name)
             throw $this->createNotFoundException("The League " . $name . " does not yet exist!");
         }
 
-        $form = $this->createForm(new LeagueForm(), $league);
+        $form = $this->createForm(new LeagueForm(), $league,array(
+            'action' => $this->get('router')->generate('editLeague',array(
```
Comment (dbeier) [inline src/AppBundle/Controller/LeagueController.php]: inject router

File: src/AppBundle/Controller/PlayerController.php [hunk 8]
```diff
@@ -21,23 +21,26 @@ public function overviewAction()
...
     /**
-     * @Route("player",name="createPlayer")
+     * @Route("/player/new",name="createPlayer")
      */
     public function createAction(Request $request)
     {
         $player = new Player();
-        $form = $this->createForm(new PlayerForm(), $player);
+        $form = $this->createForm(new PlayerForm(), $player, array(
+            'action' => $this->get('router')->generate('createPlayer')
```
Comment (dbeier) [inline src/AppBundle/Controller/PlayerController.php]: inject

File: src/AppBundle/Controller/PlayerController.php [hunk 9]
```diff
@@ -21,23 +21,26 @@ public function overviewAction()
...
+        ));
+
 
         $form->handleRequest($request);
 
         if ($form->isSubmitted()) {
             $em = $this->getDoctrine()->getManager();
             $em->persist($player);
```
Comment (dbeier) [inline src/AppBundle/Controller/PlayerController.php]: maybe work here with repository? you also cann then add a save function to the repositry which uses persist and flush

File: src/AppBundle/Controller/PlayerController.php [hunk 10]
```diff
@@ -49,35 +52,40 @@ public function lineupAction(){
...
     /**
-     * @Route("player/{name}")
+     * @Route("player/{id}", name="editPlayer")
      * @return \Symfony\Component\HttpFoundation\Response
      **/
-    public function editAction(Request $request, $name)
+    public function editAction(Request $request, $id)
     {
         $em = $this->getDoctrine()->getManager();
-        $player = $em->getRepository('AppBundle:Player')->findOneByname($name);
```
Comment (dbeier) [inline src/AppBundle/Controller/PlayerController.php]: inject

**Pull request merged**